### DISCUSSION TITLE
QSS_model.c

### DISCUSSION THREAD

**Discussion by mongibellili** [CIFASIS/qss-solver#180] [Q&A]
hello,
Could you please upload an example of what QSS_model.c would be like.
thank you for your time and consideration!

**Comment by joaquinffernandez** [comment]
Hi @mongibellili, sorry for the delay. in order to create the model instance needed by the QSS solver, you will have to define:

- The model definition.
- If the model is hybrid, the zero-crossing functions and their relative positive and negative handlers.
- The model dependencies function for state derivatives.
- The routine to initialize the engine data structures.  

The easiest way to generate the model instance is to use the MMO compiler that computes all the files needed automatically. You can either use the GUI interface or just use the compiler from the command line.
I attach an example of a model instance generated by the compiler for the [advection](https://github.com/CIFASIS/qss-solver/blob/qss-solver-dev/models/advection/advection.mo) Modelica model  

[advection.zip](https://github.com/CIFASIS/qss-solver/files/8483749/advection.zip)

**Comment by mongibellili** [comment]
Thank you. That was what I was looking for.
Keep up the good work.

**Comment by mongibellili** [comment]
Hi Joaquin and QSS solver team,
I have been following your work (articles and repository) and i have been testing and replicating some of your ideas. I was wondering if you are planning on implementing the modified Liqss mentioned in article "improving linearly implicit QSS 2018" that deals with cycles.
your help is greatly appreciated!
